### PR TITLE
(chore): bump go to 1.18

### PR DESCRIPTION
**Pull request by grokspawn**
**Description of the change:**
bump go to 1.18

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

**Review by timflannagan — APPROVED** [review]
/lgtm

**Review by njhale — COMMENTED** [review]
Looks like some empty temp files -- from a test? -- were added. Was this intentional?

File: upstream-builder.Dockerfile [hunk 1]
```diff
@@ -13,8 +13,8 @@ FROM alpine:3
 
 COPY ["nsswitch.conf", "/etc/nsswitch.conf"]
 
-RUN chgrp -R 0 /etc && \
-    chmod -R g+rwx /etc
+#RUN chgrp -R 0 /etc && \
+#    chmod -R g+rwx /etc
```
Comment (timflannagan) [inline upstream-builder.Dockerfile:17]: Should we revert these changes? These seem unrelated to the overall set of changes.

Comment (grokspawn) [reply upstream-builder.Dockerfile:17]: They're required.  (I should've deleted the lines, grumble, grumble).  /etc is on a read-only filesystem, so these commands fail.

**Review by njhale — APPROVED** [review]
Nice 👍 

/approve
